core/src/main/java/apoc/nodes/Grouping.java

```diff
@@ -368,6 +377,10 @@ private <C extends Collection<T>, T extends Entity> C fixAggregates(C pcs) {
                     double[] values = (double[]) v;
                     entry.setValue(values[1] == 0 ? 0 : values[0] / values[1]);
                 }
+                if (k.matches("^avg_.+") && v instanceof DurationValue) {
+                    Long count = ((Number) pc.getProperty(k + "_count", 0)).longValue();
```

Comment:
Is it intentional to use `.longValue()` here? I thinking about `Double`s for example, that will get rounded by that call.